osu.Framework/Graphics/Animations/Animation.cs

```diff
@@ -90,8 +141,11 @@ public void GotoFrame(int frameIndex)
             else if (frameIndex >= frameData.Count)
                 frameIndex = frameData.Count - 1;
 
-            CurrentFrameIndex = frameIndex;
-            updateCurrentFrame();
+            if (!startAtCurrentTime)
+                throw new InvalidOperationException($"A {nameof(Animation<T>)} with {startAtCurrentTime} = false cannot seek as it is dependent on an external clock.");
```

Comment:
same here (adding suggestion to speed things up & make sure nothing is missed)
```suggestion
                throw new InvalidOperationException($"A {nameof(Animation<T>)} with {nameof(startAtCurrentTime)} = false cannot seek as it is dependent on an external clock.");
```